superset/sql_lab.py

```diff
@@ -366,6 +386,10 @@ def execute_sql_statements(  # pylint: disable=too-many-arguments, too-many-loca
         # only asynchronous queries
         stats_logger.timing("sqllab.query.time_pending", now_as_float() - start_time)
 
+    if not hasattr(g, "user"):
+        # pylint: disable=assigning-non-slot
+        g.user = security_manager.find_user(username=user_name)
```

Comment:
why do we need to reset `g.user` here?

Reply:
When I was testing, `g.user` was not set here or further down the stack, so the associated RLS rules were not being fetched. I assume this happens because the `execute_sql_statements` function is running either in Celery (if the DB is async) or in a separate thread (if the DB is sync).

Reply:
Makes sense for celery, yet notice that celery tasks have a Flask App context on Superset but not a user (g.user) unless we impersonate.

Take a look at: https://github.com/apache/superset/pull/19914/files

I think that setting `g.user` should either be short lived (like on #19914) and/or done on expected code paths (on request set on security manager). I think this could lead to unexpected behaviours, since calling `execute_sql_statements` would unexpectedly mutate `g.user`

Reply:
This path ends up calling the security manager to get the RLS rules, which is why I was setting `g.user`. But I can change it to pass the username instead, would that be better?

Reply:
Done in [00c970bb432dbf513c2c2e363fb45400709e14b0](https://github.com/apache/superset/pull/19999/commits/00c970bb432dbf513c2c2e363fb45400709e14b0)